addons/hr_holidays/models/hr_leave.py

```diff
@@ -594,16 +594,17 @@ def _get_number_of_days(self, date_from, date_to, employee_id):
         """ Returns a float equals to the timedelta between two dates given as string."""
         if employee_id:
             employee = self.env['hr.employee'].browse(employee_id)
-            return employee._get_work_days_data_batch(date_from, date_to)[employee.id]
+            if not self.request_unit_half:
+                return employee._get_work_days_data_batch(date_from, date_to)[employee.id]
```

Comment:
Why not ? 
```suggestion
            if self.request_unit_half:
                return 0.5
            return employee._get_work_days_data_batch(date_from, date_to)[employee.id]
```
We are still in the case there is an employee set on the time off. 

Thanks for the patch !

Reply:
@tivisse the function must return this format ```{'days': days, 'hours': hours}``` where I guess the exact number of hours must be computed if we count the leave in hours